src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/OpenSslCachedSystemStoreProvider.cs

```diff
@@ -126,61 +147,76 @@ public void Remove(ICertificatePal cert)
             Interop.Crypto.CheckValidOpenSslHandle(intermedStore);
 
-            DateTime newFileTime = default;
-            DateTime newDirTime = default;
-
             var uniqueRootCerts = new HashSet<X509Certificate2>();
             var uniqueIntermediateCerts = new HashSet<X509Certificate2>();
             bool firstLoad = (s_nativeCollections == null);
 
-            if (rootStoreFile != null && rootStoreFile.Exists)
+            if (firstLoad)
+            {
+                s_rootStoreDirectories = GetRootStoreDirectories(out s_defaultRootDir);
+                s_directoryLastWrite = new DateTime[s_rootStoreDirectories.Length];
+                s_rootStoreFile = GetRootStoreFile();
+            }
+            else
+            {
+                Debug.Assert(s_rootStoreDirectories is not null);
+                Debug.Assert(s_directoryLastWrite is not null);
+            }
+
+            if (s_rootStoreFile != null)
             {
-                newFileTime = ContentWriteTime(rootStoreFile);
-                ProcessFile(rootStoreFile);
+                ProcessFile(s_rootStoreFile, out s_fileLastWrite);
             }
 
             bool hasStoreData = false;
 
-            if (rootStorePath != null && rootStorePath.Exists)
+            for (int i = 0; i < s_rootStoreDirectories.Length; i++)
             {
-                newDirTime = ContentWriteTime(rootStorePath);
-                hasStoreData = ProcessDir(rootStorePath);
+                hasStoreData = ProcessDir(s_rootStoreDirectories[i], out s_directoryLastWrite[i]);
             }
 
             if (firstLoad && !hasStoreData && s_defaultRootDir)
             {
-                DirectoryInfo etcSslCerts = new DirectoryInfo("/etc/ssl/certs");
-
-                if (etcSslCerts.Exists)
+                const string DefaultCertDir = "/etc/ssl/certs";
+                hasStoreData = ProcessDir(DefaultCertDir, out DateTime newDirTime);
+                if (hasStoreData)
                 {
-                    DateTime tmpTime = ContentWriteTime(etcSslCerts);
-                    hasStoreData = ProcessDir(etcSslCerts);
-
-                    if (hasStoreData)
-                    {
-                        newDirTime = tmpTime;
-                        s_rootStoreDirectoryInfo = etcSslCerts;
-                    }
+                    s_rootStoreDirectories = new[] { DefaultCertDir };
+                    s_directoryLastWrite = new[] { newDirTime };
                 }
             }
 
-            bool ProcessDir(DirectoryInfo dir)
+            bool ProcessDir(string dir, out DateTime lastModified)
             {
+                if (!TryStatDirectory(dir, out lastModified))
+                {
+                    return false;
+                }
+
                 bool hasStoreData = false;
 
-                foreach (FileInfo file in dir.EnumerateFiles())
+                foreach (string file in Directory.GetFiles(dir))
```

Comment:
Since it's just being used in a foreach, EnumerateFiles is better (GetFiles is literally `new List<string>(EnumerateFiles(...)).ToArray()`)

```suggestion
                foreach (string file in Directory.EnumerateFiles(dir))
```